docs/reference-guide/storage-modes.md

```diff
@@ -117,4 +138,119 @@ In the `IN_MEMORY_ANALYTICAL` mode, periodic snapshots are **disabled**.
 Manual snapshots are created by running the `CREATE SNAPSHOT;` query. When the
 query is run in the `IN_MEMORY_ANALYTICAL` mode, Memgraph guarantees that it
 will be **the only** transaction present in the system, and all the other
-transactions will wait until the snapshot is created to ensure its validity.
\ No newline at end of file
+transactions will wait until the snapshot is created to ensure its validity.
+
+## On-disk transactional storage mode
+
+In the on-disk transactional storage mode, disk is used as a physical storage
+which allows you to save more data than the capacity of your RAM. This helps
+keep the hardware costs to a minimum, but you should except slower performance
+when compared to the in-memory transactional storage mode. Keep in mind that
+while executing queries, all the graph objects used in the transactions still
+need to be able to fit in the RAM, or Memgraph will thrown an exception. 
+
+### Architecture
+
+RocksDB is used as a background storage to serialize nodes and relationships
+into a key-value format. The used architecture is also known as "larger than
+memory" as it enables in-memory databases to save more data than the main memory
+can hold, without the performance overhead caused by the buffer pool. 
+
+The imported data is residing on the disk, while the main memory contains two
+caches, one executing operations on main RocksDB instance and the other for
+operations that require indices. The cache used here is Memgraph's custom
+`SkipList` cache which allows multithreaded read-write access pattern.
```

Comment:
Which cache or both?

Reply:
SkipList is cache

Reply:
I mean any SkipList allows multithreaded read-write access pattern.

Reply:
So both caches used in the main memory are SkipList?

Reply:
yes